sshd.c

```diff
@@ -525,8 +525,10 @@ privsep_preauth(struct ssh *ssh)
 
 		privsep_preauth_child();
 		setproctitle("%s", "[net]");
-		if (box != NULL)
+		if (box != NULL) {
 			ssh_sandbox_child(box);
+			free(box);
```

Comment:
sent upstream, waiting for an ok.

Reply:
Damien points out that this will cause problems for other sandboxes where the sandbox has some persistent (non-kernel) state, eg the systrace sandbox since the contents of "box" is needed throughout the life of the preauth privsep supervisor process.

Ideally we'd add a ssh_sandbox_child_finish and call it near mm_send_keystate, but the "box" variable doesn't exist in that scope and getting it there would be messy.

One option would be to free "box" at the end of ssh_sandbox_child in any sandbox implementation that can do that, and do nothing for the ones that can't.  Another option would be to just ignore it, because while it is technically a leak, it's in a process that exists only during the authentication process and exits once the authentication succeeds or fails.